.changeset/large-crabs-serve.md

```diff
@@ -0,0 +1,5 @@
+---
+'xstate': patch
+---
+
+A race condition occurred when a grandchild service is immediately stopped and the child service tried to remove it from its undefined state; this has been fixed, and the race condition no longer occurs. See https://github.com/statelyai/xstate/issues/2507 for details.
```

Comment:
As mentioned in other comments - the fixed situation was not unique to setups with grandchildren etc so a little bit more general description would be more appropriate here.